scripts/ifAlias

```diff
@@ -73,42 +97,77 @@ interface_id()
 
 interface_name()
 {
-    local IFACE_ID="$1"
-    echo "$IP_LINK" | grep -oE "^${IFACE_ID}: [^:@ ]*" | cut -d " " -f 2
+    if [ "${UNAME}" = 'Linux' ]; then
+        echo "$IP_LINK" | grep -oE "^$1: [^:@ ]*" | cut -d " " -f 2
+    else
+        echo "${IFCONFIG}" | grep -E '^[A-Za-z]+' | cut -d: -f1 | head -n "$1" | tail -n 1
+    fi
 }
 
 alias_from_interfaces_config_file()
 {
-    local IFACE="$1"
-    local CONFIG_FILE=
+    CONFIG_FILE=
 
-    if [ -x "$DISTRO_BIN" ]
-    then
-        case "$("$DISTRO_BIN" | cut -d " " -f 1)" in
+    if [ -x "$DISTRO_BIN" ]; then
+        if [ "${UNAME}" = 'Linux' ]; then
+            DISTRO_VAR="$($DISTRO_BIN | cut -d " " -f 1)"
+        else
+            DISTRO_VAR="${UNAME}"
+        fi
+
+        case "${DISTRO_VAR}" in
             Debian)
                 CONFIG_FILE="/etc/network/interfaces"
             ;;
+            Ubuntu)
+                CONFIG_FILE="/etc/network/interfaces"
+            ;;
             Gentoo)
                 CONFIG_FILE="/etc/conf.d/net"
             ;;
             CentOS|RedHat|SuSE|Mandriva|Mandrake)
-                CONFIG_FILE="/etc/sysconfig/network-scripts/ifcfg-$IFACE"
+                CONFIG_FILE="/etc/sysconfig/network-scripts/ifcfg-$1"
             ;;
             Archlinux)
-                CONFIG_FILE="/etc/conf.d/net-conf-$IFACE"
+                CONFIG_FILE="/etc/conf.d/net-conf-$1"
+            ;;
+            BSD)
+                CONFIG_FILE="/etc/rc.conf"
             ;;
         esac
     fi
-    if [ "$CONFIG_FILE" ]
-    then
-         echo "$(grep -i "^# $IFACE:" $CONFIG_FILE | sed "s/^# $IFACE: //i")"
+    if [ "$CONFIG_FILE" ]; then
+         # echo squashes possible multi line replies to a single line
+         FOUND_LINES="$(grep -i "^# $1:" $CONFIG_FILE | sed "s/^# $1: //i")"
+         if [ "$(echo "${FOUND_LINES}" | wc -l)" -ge 1 ]; then
+             echo "${FOUND_LINES}"
+             return 0
+         fi
+    fi
+    if [ "${UNAME}" = "Linux" ] && [ -d '/etc/network/interfaces.d' ]; then
+         if [ "$(/bin/ls /etc/network/interfaces.d/ | wc -l)" -ge 1 ]; then
```

Comment:
Better use find instead of ls

Reply:
Both work, pointless change made though. That said the next line did have an issue with grep.